pallets/creators-staking/src/types.rs

```diff
@@ -0,0 +1,417 @@
+/// picture of **unclaimed eras** and stakes can be constructed.
+///
+/// # Example
+/// For simplicity, the following example will represent `EraStake` using `<era, stake>` notation.
+/// Let us assume we have the following vector in `StakerInfo` struct.
+///
+/// `[<5, 1000>, <6, 1500>, <8, 2100>, <9, 0>, <11, 500>]`
+///
+/// This tells us which eras are unclaimed and how much it was staked in each era.
+/// The interpretation is the following:
+/// 1. In era **5**, staked amount was **1000** (interpreted from `<5, 1000>`)
+/// 2. In era **6**, staker staked additional **500**, increasing total staked amount to **1500**
+/// 3. No entry for era **7** exists which means there were no changes from the former entry.
+///    This means that in era **7**, staked amount was also **1500**
+/// 4. In era **8**, staker staked an additional **600**, increasing total stake to **2100**
+/// 5. In era **9**, staker unstaked everything from the creator (interpreted from `<9, 0>`)
+/// 6. No changes were made in era **10** so we can interpret this same as the previous entry which
+/// means **0** staked amount. 7. In era **11**, staker staked **500** on the creator, making his
+/// stake active again after 2 eras of inactivity.
+///
+/// **NOTE:** It is important to understand that staker **DID NOT** claim any rewards during this
+/// period.
+#[derive(Encode, Decode, Clone, Default, PartialEq, Eq, RuntimeDebug, TypeInfo)]
+pub struct StakerInfo<Balance: AtLeast32BitUnsigned + Copy + MaxEncodedLen> {
+    // Size of this list would be limited by a configurable constant
+    stakes: Vec<EraStake<Balance>>,
+}
+
+impl<Balance: AtLeast32BitUnsigned + Copy + MaxEncodedLen> MaxEncodedLen for StakerInfo<Balance> {
+    // This is just an assumption, will be calculated properly in the future. See the comment for
+    // `MAX_ASSUMED_VEC_LEN`.
+    fn max_encoded_len() -> usize {
+        codec::Compact(MAX_ASSUMED_VEC_LEN).encoded_size().saturating_add(
+            (MAX_ASSUMED_VEC_LEN as usize).saturating_mul(EraStake::<Balance>::max_encoded_len()),
+        )
+    }
+}
+
+impl<Balance: AtLeast32BitUnsigned + Copy + MaxEncodedLen> StakerInfo<Balance> {
+    /// `true` if no active stakes and unclaimed eras exist, `false` otherwise
+    pub(super) fn is_empty(&self) -> bool {
+        self.stakes.is_empty()
+    }
+
+    /// number of `EraStake` chunks
+    pub(super) fn len(&self) -> u32 {
+        self.stakes.len() as u32
+    }
+
+    /// Stakes some value in the specified era.
+    ///
+    /// User should ensure that given era is either equal or greater than the
+    /// latest available era in the staking info.
+    ///
+    /// # Example
+    ///
+    /// The following example demonstrates how internal vector changes when `stake` is called:
+    ///
+    /// `stakes: [<5, 1000>, <7, 1300>]`
+    /// * `stake(7, 100)` will result in `[<5, 1000>, <7, 1400>]`
+    /// * `stake(9, 200)` will result in `[<5, 1000>, <7, 1400>, <9, 1600>]`
```

Comment:
<img width="464" alt="Знімок екрана 2023-09-05 о 12 55 20" src="https://github.com/dappforce/subsocial-parachain/assets/32250097/5a2e9cd5-202d-4688-9126-2b9b8564d53f">

Because of this. `stake` function adds the specified value to the already staked amount. That's why it creates a new record with a resulting stake.